src/state/routing/usePreviewTrade.ts

```diff
@@ -0,0 +1,134 @@
+import { skipToken } from '@reduxjs/toolkit/query/react'
+import { ChainId, Currency, CurrencyAmount, Percent, TradeType } from '@uniswap/sdk-core'
+import { ZERO_PERCENT } from 'constants/misc'
+import { useQuickRouteMainnetEnabled } from 'featureFlags/flags/quickRouteMainnet'
+import { useMemo } from 'react'
+
+import { useGetQuickRouteQuery, useGetQuickRouteQueryState } from './quickRouteSlice'
+import { GetQuickQuoteArgs, PreviewTrade, QuoteState, TradeState } from './types'
+import { currencyAddressForSwapQuote } from './utils'
+
+const TRADE_NOT_FOUND = { state: TradeState.NO_ROUTE_FOUND, trade: undefined } as const
+const TRADE_LOADING = { state: TradeState.LOADING, trade: undefined } as const
+
+function useQuickRouteArguments({
+  tokenIn,
+  tokenOut,
+  amount,
+  tradeType,
+  inputTax,
+  outputTax,
+}: {
+  tokenIn?: Currency
+  tokenOut?: Currency
+  amount?: CurrencyAmount<Currency>
+  tradeType: TradeType
+  inputTax: Percent
+  outputTax: Percent
+}): GetQuickQuoteArgs | typeof skipToken {
+  const enabledMainnet = useQuickRouteMainnetEnabled()
+
+  return useMemo(() => {
+    if (!tokenIn || !tokenOut || !amount) return skipToken
+    if (tokenIn.chainId !== ChainId.MAINNET || (tokenIn.chainId === ChainId.MAINNET && !enabledMainnet))
+      return skipToken
+
+    return {
+      amount: amount.quotient.toString(),
+      tokenInAddress: currencyAddressForSwapQuote(tokenIn),
+      tokenInChainId: tokenIn.chainId,
+      tokenInDecimals: tokenIn.wrapped.decimals,
+      tokenInSymbol: tokenIn.wrapped.symbol,
+      tokenOutAddress: currencyAddressForSwapQuote(tokenOut),
+      tokenOutChainId: tokenOut.wrapped.chainId,
+      tokenOutDecimals: tokenOut.wrapped.decimals,
+      tokenOutSymbol: tokenOut.wrapped.symbol,
+      tradeType,
+      inputTax,
+      outputTax,
+    }
+  }, [amount, enabledMainnet, inputTax, outputTax, tokenIn, tokenOut, tradeType])
+}
+/**
+ * Returns the best trade by invoking the routing api or the smart order router on the client
+ * @param tradeType whether the swap is an exact in/out
+ * @param amountSpecified the exact amount to swap in/out
+ * @param otherCurrency the desired output/payment currency
```

Comment:
I believe this comment was copied and pasted, would be worth updating since this trade explicitly doesn't retrieve the best trade